src/main/kotlin/pl/edu/agh/gem/internal/factory/EmailFactory.kt

```diff
@@ -1,7 +1,8 @@
 package pl.edu.agh.gem.internal.factory
 
 import jakarta.mail.internet.MimeMessage
+import pl.edu.agh.gem.internal.model.Attachment
 
 interface EmailFactory {
-    fun createEmail(email: String, subject: String, html: String): MimeMessage
+    fun createEmail(email: String, subject: String, html: String, attachment: Attachment? = null): MimeMessage
```

Comment:
we want to have in interface default value as null and not have that part in implementation ?